### DISCUSSION TITLE
Github Actions CI

### DISCUSSION THREAD

**Discussion by SagiPolaczek** [BiomedSciAI/fuse-med-ml#130] [Announcements]
Hey Fusers 👋 🔬 

We now support Github actions CI - meaning each PR will have to pass additional tests focusing on style and format to ensure uniform code style in Fuse.
For your convenience here is a quick recap:

We use the following three tools: [`black`](https://github.com/psf/black), [`flake8`](https://flake8.pycqa.org/en/latest/) and [`mypy`](https://mypy.readthedocs.io/en/stable/).

### `black`
Enforces code style and format. We use the standard style guideline with a minor change: we restrict lines' lengths to be 120 (instead of 88).

To check if file / directory suit the code style guidelines and see the changes `black` will make:
```sh
black <path> --check --diff --color
```
To apply the changes:
```sh
black <path>
```

### `flake8`
Checks code style (similar to black)  and detects various issues not covered by black.
usage:
```sh
flake8 <path>
```
**TIP** : If `flake8` gives you a lot of error messages, use `black`'s formatter might solve most of them. The rest should be done manually.

###  `mypy`
Enforces the usage of type annotations for functions input arguments and return values only, but not the correctness of them. For examples:
```python
def this_func_returns_int(number): # BAD
    ...
def this_func_returns_int(number: int): # BAD
    ...
def this_func_returns_int(number: int) -> int: # GOOD
    ...
def this_func_returns_int(number: str) -> str: # Will pass, but not so good
    ...
```

You are welcome to ask questions and raise issues in the discussion thread below.
Happy Fusing 🔬

**Comment by SagiPolaczek** [comment]
@mosheraboh @michalozeryflato @YoelShoshan @alex-golts @itaijj @liamhazan @egozi

**Reply by mosheraboh** [reply]
Thanks Sagi!
Sending more notifications :)
@afoncubierta @akoseki2 @vagenas @simona-rc @floccinauc @shakedpe @vesnarb @efratHex
